feat: introduce new centralized OTLP gateway architecture (#3148)
kyma-project/telemetry-manager: docs/user/architecture/README.md

1. Watch the module configuration for changes and sync the module status to it.
2. Watch the user-created Kubernetes resources LogPipeline, TracePipeline, and MetricPipeline. In these resources, you specify what data of a signal type to collect and where to ship it.
3. Manage the lifecycle of the self monitor, the OTLP Gateway, and the signal-specific agents (that is, the Log Agent and Metric Agent).
   The OTLP Gateway is deployed when you create any pipeline resource (LogPipeline, TracePipeline, or MetricPipeline).

![Manager](./../assets/manager-resources.drawio.svg)

## OTLP Gateway and Agents

The OTLP Gateway and signal-specific agents (that is, the Log Agent and Metric Agent) handle the incoming telemetry data. Telemetry Manager deploys them based on your pipeline configuration.

> The OTLP Gateway is based on an [OTel Collector](https://opentelemetry.io/docs/collector/) [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) running one instance per cluster node. It acts as the central endpoint to which your applications push telemetry data in the OTLP format. The gateway enriches and filters the data, and then dispatches it to the backends configured in your pipeline resources. The gateway handles all signal types (logs, traces, and metrics) in a single unified component.

Reviewer: How about this - with the "enrich" bit and replacing the "distributed gateway" phrase:

```suggestion
The OTLP Gateway acts as the central endpoint where your applications push telemetry data (logs, traces, and metrics) using the OTLP format. It enriches and filters this data, then dispatches it to the backends you configure in your pipeline resources. The gateway runs as an [OTel Collector](https://opentelemetry.io/docs/collector/)-based [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) with one instance per cluster node.
```

- Traces: `http://telemetry-otlp-traces.kyma-system:4317` (gRPC) or `http://telemetry-otlp-traces.kyma-system:4318` (HTTP)
- Metrics: `http://telemetry-otlp-metrics.kyma-system:4317` (gRPC) or `http://telemetry-otlp-metrics.kyma-system:4318` (HTTP)
- Logs: `http://telemetry-otlp-logs.kyma-system:4317` (gRPC) or `http://telemetry-otlp-logs.kyma-system:4318` (HTTP)

Agents run as [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) and pull data from the respective node.

- **Log Agent**: Collects logs from the stdout/stderr output of all containers on a node. For details, see [Logs Architecture](./logs-architecture.md). As an alternative to the OTLP-based log feature, you can use a log agent based on a [Fluent Bit](https://fluentbit.io/) installation running as a DaemonSet. It reads all containers’ logs in the runtime and ships them according to your LogPipeline configuration. For details, see [Application Logs (Fluent Bit)](./../02-logs.md).
- **Metric Agent**: Scrapes Prometheus-annotated workloads on each node. For details, see [Metrics Architecture](./metrics-architecture.md).

## Self Monitor

> The Telemetry module includes a [Prometheus](https://prometheus.io/)-based self-monitor that collects and evaluates health metrics from the OTLP Gateway and agents. Telemetry Manager uses this data to report the current health status in your pipeline resources.

Reviewer: ```suggestion
The Telemetry module includes a [Prometheus](https://prometheus.io/)-based self-monitor that collects and evaluates health metrics from the OTLP Gateway and signal-specific agents. Telemetry Manager uses this data to report the current health status in your pipeline resources.
```
